californium-core/src/main/java/org/eclipse/californium/core/network/UdpMatcher.java

```diff
@@ -508,11 +529,12 @@ private KeyToken createUnusedToken() {
 				KeyMID idByMID = new KeyMID(exchange.getCurrentRequest().getMID());
 				KeyToken idByToken = new KeyToken(exchange.getCurrentRequest().getToken());
 
-				exchangesByToken.remove(idByToken);
-
-				// in case an empty ACK was lost
-				exchangesByMID.remove(idByMID);
+				if (!exchange.getRequest().isMulticast() || isMulticastTimedOut(exchange)) {
+					exchangesByToken.remove(idByToken);
```

Comment:
Should the cleanup task be scheduled in `sendRequest(Exchange, Request)` method of `UdpMatcher`?

Where to get the `MAX_LATENCY` and `MAX_SERVER_RESPONSE_DELAY` values? Looks like `NetworkConfig` doesn't have such keys.